packages/@aws-cdk/aws-ec2/lib/vpc-endpoint.ts

```diff
@@ -252,6 +253,42 @@ export class InterfaceVpcEndpointService implements IInterfaceVpcEndpointService
   }
 }
 
+// The following services need to have their interface endpoints prefixed with 'cn' (before com.amazonaws.{REGION}) for china regions
+const CN_OVERRIDES = ['application-autoscaling',
```

Comment:
This list will be hard to maintain, is this documented anywhere? 
Questions/Considerations:

1. Can we simply add the partition regardless of the service and partition? Using `Aws.PARTITION` .
2. If the partition is required only in the china partition, can we only append it if the stack is created in the china partition?

Reply:
There are two regions(cn-north-1, cn-northwest-1) having the different service endpoints, my comments in #9864 : https://github.com/aws/aws-cdk/issues/9864#issuecomment-897395089 , my draft idea is to add a method getDefaultPrefix() in 
> aws-cdk/packages/@aws-cdk/aws-ec2/lib/vpc-endpoint.ts:

``` typescript
this.name = `${prefix || 'com.amazonaws'}.${Aws.REGION}.${name}` 
==>  this.name = `${prefix || getDefaultPrefix()}.${Aws.REGION}.${name}`
```

Do you have any suggestions on this?

Reply:
This should still work though because there are no services where the cn-north-1 endpoint is different than cn-northwest-1 endpoint for the same service.. The list may not be comprehensive and from your list it looks like maybe there is support for iotsitewise endpoints only in cn-north-1 and only for workspaces in cn-northwest-1. So to address that we could take the union of the 2 lists. As the reviewer mentioned though it may be difficult to maintain. I'm not sure if there is a way to invoke the ec2 api from within this repo but that's the only way I can think of to have the most up to date info.